core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -633,18 +643,20 @@ public void deleteShardStore(String reason, ShardId shardId, ClusterState cluste
     }
 
     /**
-     * This method returns true if the current node is allowed to delete the
-     * given index. If the index uses a shared filesystem this method always
-     * returns false.
+     * This method returns true if the current node is allowed to delete the given index.
+     * This is the case if the index is deleted in the metadata or there is no allocation
+     * on the local node and the index isn't on a shared file system.
      * @param index {@code Index} to check whether deletion is allowed
      * @param indexSettings {@code IndexSettings} for the given index
+     * @param closed true if the index has been deleted closed already, but not yet deleted
      * @return true if the index can be deleted on this node
      */
     public boolean canDeleteIndexContents(Index index, IndexSettings indexSettings, boolean closed) {
-        final IndexService indexService = indexService(index);
-        // Closed indices may be deleted, even if they are on a shared
-        // filesystem. Since it is closed we aren't deleting it for relocation
+        // index contents can be deleted if the index is not on a shared file system,
+        // or if its on a shared file system but its an already closed index (so all
+        // its resources have already been relinquished)
         if (indexSettings.isOnSharedFilesystem() == false || closed) {
```

Comment:
since we have the metadata here (as part of IndexSettings), can we check here if the index is closed? this means we can remove the boolean parameter and not have to worry about how we got here